### PR TITLE
Skill.from_config accepts agent context instead of skill context.

### PR DESCRIPTION
**Pull request by marcofavorito**
## Proposed changes

make Skill.from_config and from_dir accept agent context instead of skill context.

## Fixes

related to  #1178 and  #1290

## Types of changes

What types of changes does your code introduce to agents-aea?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply._

- [ ] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that code coverage does not decrease.
- [ ] I have checked that the documentation about the `aea cli` tool works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Further comments

n/a

### CODE REVIEW

File: tests/test_aea.py [hunk 1]
```diff
@@ -348,23 +347,24 @@ def test_initialize_aea_programmatically_build_resources():
...
+            dummy_skill = Skill.from_dir(
+                str(Path(CUR_PATH, "data", "dummy_skill")), agent_context=aea.context
+            )
             resources.add_skill(dummy_skill)
             resources.add_skill(error_skill)
 
-            aea = AEA(identity, connections, wallet, ledger_apis, resources=resources,)
-
             error_skill.skill_context.set_agent_context(aea.context)
             dummy_skill.skill_context.set_agent_context(aea.context)
```
Comment (DavidMinarsch) [inline tests/test_aea.py:367]: No longer needed?

File: aea/skills/base.py [hunk 2]
```diff
@@ -627,30 +627,31 @@ def models(self) -> Dict[str, Model]:
...
             ComponentConfiguration.load(ComponentType.SKILL, Path(directory)),
         )
         configuration._directory = Path(directory)
-        return Skill.from_config(configuration, skill_context)
+        return Skill.from_config(configuration, agent_context)
 
     @classmethod
     def from_config(
-        cls, configuration: SkillConfig, skill_context: Optional[SkillContext] = None
+        cls, configuration: SkillConfig, agent_context: Optional[AgentContext] = None
```
Comment (DavidMinarsch) [inline aea/skills/base.py:648]: Why not make it required? What is the usecase stopping us from doing that?

File: aea/aea_builder.py [hunk 3]
```diff
@@ -1211,33 +1212,10 @@ def _load_and_add_components(
                 component = self._component_instances[component_type][configuration]
             else:
                 configuration = deepcopy(configuration)
-                component = load_component_from_config(component_type, configuration)
-            resources.add_component(component)
-
-    def _load_and_add_skills(self, context: AgentContext, resources: Resources) -> None:
```
Comment (DavidMinarsch) [inline aea/aea_builder.py:1217]: Awesome!

File: aea/skills/base.py (hunk 2 above)
Comment (marcofavorito) [reply aea/skills/base.py]: done

File: tests/test_aea.py (hunk 1 above)
Comment (marcofavorito) [reply tests/test_aea.py]: yes, done

**Pull request merged**